app/logic/Mail/OWA/OWAAccount.ts

```diff
@@ -230,13 +253,21 @@ export class OWAAccount extends MailAccount {
     await this.callOWA(request);
   }
 
-  async callOWA(aRequest: any): Promise<any> {
+  async callOWA(aRequest: any, aMailbox?: string): Promise<any> {
+    if (this.mainAccount) {
+      return await (this.mainAccount as OWAAccount).callOWA(aRequest, this.username);
+    }
     if (!this.hasLoggedIn) {
       throw new LoginError(null, "Please login");
     }
     let url = this.url + 'service.svc';
     let options = {
-      headers: {
+      headers: aMailbox ? {
+        Action: aRequest.action,
+        "x-anchormailbox": aMailbox,
+        "x-customowascenariodata": "MailboxAccess:SharedMailbox,ExplicitLogon",
+        "x-owa-explicitlogonuser": aMailbox,
```

Comment:
So both these are the shared user, not my user? explicitlogonuser suggests to me that this is the user I logged in with, i.e. my user, not the shared user.